# Data cleaning project

This is a simple data cleaning project which I did when I started learning Python.
The dataset is called 'trending youtube videos statistic' which you can find in Kaggle website. I chose to do only the first file which is the GBvideos.csv.
Towards the end of this notebook, I also did some simple data extraction. 

In [1]:
#Import libraries

import numpy as np 
import pandas as pd 

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory
import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

/kaggle/input/youtube-new/GBvideos.csv
/kaggle/input/youtube-new/MXvideos.csv
/kaggle/input/youtube-new/KRvideos.csv
/kaggle/input/youtube-new/MX_category_id.json
/kaggle/input/youtube-new/GB_category_id.json
/kaggle/input/youtube-new/US_category_id.json
/kaggle/input/youtube-new/IN_category_id.json
/kaggle/input/youtube-new/DEvideos.csv
/kaggle/input/youtube-new/KR_category_id.json
/kaggle/input/youtube-new/RU_category_id.json
/kaggle/input/youtube-new/FRvideos.csv
/kaggle/input/youtube-new/USvideos.csv
/kaggle/input/youtube-new/INvideos.csv
/kaggle/input/youtube-new/RUvideos.csv
/kaggle/input/youtube-new/CA_category_id.json
/kaggle/input/youtube-new/DE_category_id.json
/kaggle/input/youtube-new/JP_category_id.json
/kaggle/input/youtube-new/JPvideos.csv
/kaggle/input/youtube-new/FR_category_id.json
/kaggle/input/youtube-new/CAvideos.csv


In [2]:
#Load the Data into Pandas Dataframe
youtube = pd.read_csv("/kaggle/input/youtube-new/GBvideos.csv")

In [3]:
#view the first five rows of the dataframe
youtube.head()

video_id trending_date  \
0  Jw1Y-zhQURU      17.14.11   
1  3s1rvMFUweQ      17.14.11   
2  n1WpP7iowLc      17.14.11   
3  PUTEiSjKwJU      17.14.11   
4  rHwDegptbI4      17.14.11   

                                               title  \
0      John Lewis Christmas Ad 2017 - #MozTheMonster   
1          Taylor Swift: …Ready for It? (Live) - SNL   
2         Eminem - Walk On Water (Audio) ft. Beyoncé   
3  Goals from Salford City vs Class of 92 and Fri...   
4  Dashcam captures truck's near miss with child ...   

                channel_title  category_id              publish_time  \
0                  John Lewis           26  2017-11-10T07:38:29.000Z   
1         Saturday Night Live           24  2017-11-12T06:24:44.000Z   
2                  EminemVEVO           10  2017-11-10T17:00:03.000Z   
3  Salford City Football Club           17  2017-11-13T02:30:38.000Z   
4            Cute Girl Videos           25  2017-11-13T01:45:13.000Z   

                                                tags     views   likes  \
0  christmas|"john lewis christmas"|"john lewis"|...   7224515   55681   
1  SNL|"Saturday Night Live"|"SNL Season 43"|"Epi...   1053632   25561   
2  Eminem|"Walk"|"On"|"Water"|"Aftermath/Shady/In...  17158579  787420   
3  Salford City FC|"Salford City"|"Salford"|"Clas...     27833     193   
4                                             [none]      9815      30   

   dislikes  comment_count                                  thumbnail_link  \
0     10247           9479  https://i.ytimg.com/vi/Jw1Y-zhQURU/default.jpg   
1      2294           2757  https://i.ytimg.com/vi/3s1rvMFUweQ/default.jpg   
2     43420         125882  https://i.ytimg.com/vi/n1WpP7iowLc/default.jpg   
3        12             37  https://i.ytimg.com/vi/PUTEiSjKwJU/default.jpg   
4         2             30  https://i.ytimg.com/vi/rHwDegptbI4/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  Click here to continue the story and make your...  
1  Musical guest Taylor Swift performs …Ready for...  
2  Eminem's new track Walk on Water ft. Beyoncé i...  
3  Salford drew 4-4 against the Class of 92 and F...  
4  Dashcam captures truck's near miss with child ...

In [4]:
#number of rows and features in the data and each of the features types
youtube.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 38916 entries, 0 to 38915
Data columns (total 16 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   video_id                38916 non-null  object
 1   trending_date           38916 non-null  object
 2   title                   38916 non-null  object
 3   channel_title           38916 non-null  object
 4   category_id             38916 non-null  int64 
 5   publish_time            38916 non-null  object
 6   tags                    38916 non-null  object
 7   views                   38916 non-null  int64 
 8   likes                   38916 non-null  int64 
 9   dislikes                38916 non-null  int64 
 10  comment_count           38916 non-null  int64 
 11  thumbnail_link          38916 non-null  object
 12  comments_disabled       38916 non-null  bool  
 13  ratings_disabled        38916 non-null  bool  
 14  video_error_or_removed  38916 non-null  bool  
 15  de

This youtube dataset has 38916 rows and 16 columns i.e features.

In [5]:
#number of missing value in youtube dataset
missing_values_count = youtube.isnull().sum()
missing_values_count

#only description column has missing values

video_id                    0
trending_date               0
title                       0
channel_title               0
category_id                 0
publish_time                0
tags                        0
views                       0
likes                       0
dislikes                    0
comment_count               0
thumbnail_link              0
comments_disabled           0
ratings_disabled            0
video_error_or_removed      0
description               612
dtype: int64

In [6]:
#percentage of missing values 

# how many total missing values do we have?
total_cells = np.product(youtube.shape)
total_missing = missing_values_count.sum()

# percent of data that is missing
(total_missing/total_cells) * 100

0.0982886216466235

In [10]:
#one method to find the index of null values to check if we have correctly replace the missing values
#can also just check df.isnull().sum()
youtube.loc[pd.isna(youtube["description"]), :].index

#another method
#indices = list(np.where(youtube['description'].isnull())[0])

Int64Index([   91,   118,   187,   309,   336,   394,   521,   546,   760,
              787,
            ...
            38220, 38283, 38339, 38387, 38457, 38511, 38560, 38639, 38735,
            38815],
           dtype='int64', length=612)

Only the description column has null values.
We can either drop the whole column if the null values percentage is high or drop only the rows where there is null values.

Another technique to handle missing values is to replace the missing values or to impute them for numerical data.

For this project, I will replace the missing values with "no description available"

In [12]:
# replace the null values with "no description available for this video"
youtube = youtube.fillna("no description available for this video")

#check at index 91
youtube.loc[91]

video_id                                                     _wM_jY_rass
trending_date                                                   17.14.11
title                                                  Bone on Labour HQ
channel_title                                              Ross Kempsell
category_id                                                           25
publish_time                                    2017-11-06T16:12:58.000Z
tags                                                              [none]
views                                                             159065
likes                                                               1160
dislikes                                                            1164
comment_count                                                          0
thumbnail_link            https://i.ytimg.com/vi/_wM_jY_rass/default.jpg
comments_disabled                                                   True
ratings_disabled                                   

In [13]:
#youtube.isnull().sum() to check missing values after replaced
#can also use youtube.info()
youtube.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 38916 entries, 0 to 38915
Data columns (total 16 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   video_id                38916 non-null  object
 1   trending_date           38916 non-null  object
 2   title                   38916 non-null  object
 3   channel_title           38916 non-null  object
 4   category_id             38916 non-null  int64 
 5   publish_time            38916 non-null  object
 6   tags                    38916 non-null  object
 7   views                   38916 non-null  int64 
 8   likes                   38916 non-null  int64 
 9   dislikes                38916 non-null  int64 
 10  comment_count           38916 non-null  int64 
 11  thumbnail_link          38916 non-null  object
 12  comments_disabled       38916 non-null  bool  
 13  ratings_disabled        38916 non-null  bool  
 14  video_error_or_removed  38916 non-null  bool  
 15  de

In [14]:
#find complete duplicates

column_names = ['title','trending_date']
duplicates2 = youtube.duplicated(subset = column_names, keep=False)
youtube[duplicates2].sort_values(by='title')

#there are 348 rows with duplicates (same video and same trending date and other columns also same)

video_id trending_date  \
34137  0XuOkYlUWIs      18.16.05   
33989  0XuOkYlUWIs      18.16.05   
33829  0XuOkYlUWIs      18.15.05   
33903  0XuOkYlUWIs      18.15.05   
33972  pojXEFDFMw8      18.16.05   
...            ...           ...   
33642  RECFTvCXhPk      18.14.05   
33662  ao_CJr7-NR4      18.14.05   
33776  ao_CJr7-NR4      18.14.05   
33784  PcuRZ3WACBU      18.14.05   
33670  PcuRZ3WACBU      18.14.05   

                                                   title  \
34137  13 Reasons Why: Season 2 | Official Trailer [H...   
33989  13 Reasons Why: Season 2 | Official Trailer [H...   
33829  13 Reasons Why: Season 2 | Official Trailer [H...   
33903  13 Reasons Why: Season 2 | Official Trailer [H...   
33972                 2018 FIFA World Cup | Forget | ITV   
...                                                  ...   
33642  will.i.am Finds Kanye West's Comments About Sl...   
33662  ‘Karate Kid’ Sequel ‘Cobra Kai’ Reunites Ralph...   
33776  ‘Karate Kid’ Sequel ‘Cobra Kai’ Reunites Ralph...   
33784  🔥  'Avengers: Infinity War' Cast Answer Your B...   
33670  🔥  'Avengers: Infinity War' Cast Answer Your B...   

              channel_title  category_id              publish_time  \
34137               Netflix           24  2018-05-08T14:00:00.000Z   
33989               Netflix           24  2018-05-08T14:00:00.000Z   
33829               Netflix           24  2018-05-08T14:00:00.000Z   
33903               Netflix           24  2018-05-08T14:00:00.000Z   
33972                   ITV           24  2018-05-11T11:00:09.000Z   
...                     ...          ...                       ...   
33642  Good Morning Britain           24  2018-05-02T10:55:44.000Z   
33662                 TODAY           25  2018-05-02T15:00:05.000Z   
33776                 TODAY           25  2018-05-02T15:00:05.000Z   
33784              MTV News           24  2018-04-29T16:00:02.000Z   
33670              MTV News           24  2018-04-29T16:00:02.000Z   

                                                    tags    views   likes  \
34137  Netflix|"Trailer"|"Netflix Original Series"|"N...  6976929  233115   
33989  Netflix|"Trailer"|"Netflix Original Series"|"N...  6976929  233115   
33829  Netflix|"Trailer"|"Netflix Original Series"|"N...  6760593  229250   
33903  Netflix|"Trailer"|"Netflix Original Series"|"N...  6760593  229250   
33972  tv|"television"|"ITV"|"Britain"|"British"|"UK"...   248238    3454   
...                                                  ...      ...     ...   
33642  good morning britain|"breakfast show"|"news"|"...  1825063   31944   
33662  The TODAY Show|"TODAY Show"|"TODAY"|"NBC"|"NBC...  1054267   12339   
33776  The TODAY Show|"TODAY Show"|"TODAY"|"NBC"|"NBC...  1054267   12339   
33784  mtv news|"celebrity news"|"entertainment news"...   854975   16647   
33670  mtv news|"celebrity news"|"entertainment news"...   854975   16647   

       dislikes  comment_count  \
34137      9027          22528   
33989      9027          22528   
33829      8891          22123   
33903      8891          22123   
33972       394              0   
...         ...            ...   
33642      4045              0   
33662       383           2306   
33776       383           2306   
33784       192            551   
33670       192            551   

                                       thumbnail_link  comments_disabled  \
34137  https://i.ytimg.com/vi/0XuOkYlUWIs/default.jpg              False   
33989  https://i.ytimg.com/vi/0XuOkYlUWIs/default.jpg              False   
33829  https://i.ytimg.com/vi/0XuOkYlUWIs/default.jpg              False   
33903  https://i.ytimg.com/vi/0XuOkYlUWIs/default.jpg              False   
33972  https://i.ytimg.com/vi/pojXEFDFMw8/default.jpg               True   
...                                               ...                ...   
33642  https://i.ytimg.com/vi/RECFTvCXhPk/default.jpg               True   
33662  https://i.ytimg.com/vi/ao_CJr7-NR4/default.jpg              False 

In [15]:
youtube['title'].loc[34137]

'13 Reasons Why: Season 2 | Official Trailer [HD] | Netflix'

In [16]:
#drop duplicates of column title and trending date and keep the first entry only
youtube = youtube.drop_duplicates(['title','trending_date'],keep= 'first')

#check if we have successfully remove duplicates. i.e should not have same trending date for each title
youtube[youtube['title'] == '13 Reasons Why: Season 2 | Official Trailer [HD] | Netflix' ]

video_id trending_date  \
32808  0XuOkYlUWIs      18.09.05   
32972  0XuOkYlUWIs      18.10.05   
33144  0XuOkYlUWIs      18.11.05   
33314  0XuOkYlUWIs      18.12.05   
33489  0XuOkYlUWIs      18.13.05   
33829  0XuOkYlUWIs      18.15.05   
33989  0XuOkYlUWIs      18.16.05   
34195  0XuOkYlUWIs      18.17.05   
34359  0XuOkYlUWIs      18.18.05   
34528  0XuOkYlUWIs      18.19.05   
34692  0XuOkYlUWIs      18.20.05   
34854  0XuOkYlUWIs      18.21.05   
35009  0XuOkYlUWIs      18.22.05   
35164  0XuOkYlUWIs      18.23.05   
35323  0XuOkYlUWIs      18.24.05   
35484  0XuOkYlUWIs      18.25.05   
35655  0XuOkYlUWIs      18.26.05   
35830  0XuOkYlUWIs      18.27.05   
36003  0XuOkYlUWIs      18.28.05   
36171  0XuOkYlUWIs      18.29.05   
36340  0XuOkYlUWIs      18.30.05   

                                                   title channel_title  \
32808  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
32972  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
33144  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
33314  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
33489  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
33829  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
33989  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
34195  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
34359  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
34528  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
34692  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
34854  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
35009  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
35164  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
35323  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
35484  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
35655  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
35830  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
36003  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
36171  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   
36340  13 Reasons Why: Season 2 | Official Trailer [H...       Netflix   

       category_id              publish_time  \
32808           24  2018-05-08T14:00:00.000Z   
32972           24  2018-05-08T14:00:00.000Z   
33144           24  2018-05-08T14:00:00.000Z   
33314           24  2018-05-08T14:00:00.000Z   
33489           24  2018-05-08T14:00:00.000Z   
33829           24  2018-05-08T14:00:00.000Z   
33989           24  2018-05-08T14:00:00.000Z   
34195           24  2018-05-08T14:00:00.000Z   
34359           24  2018-05-08T14:00:00.000Z   
34528           24  2018-05-08T14:00:00.000Z   
34692           24  2018-05-08T14:00:00.000Z   
34854           24  2018-05-08T14:00:00.000Z   
35009           24  2018-05-08T14:00:00.000Z   
35164           24  2018-05-08T14:00:00.000Z   
35323           24  2018-05-08T14:00:00.000Z   
35484           24  2018-05-08T14:00:00.000Z   
35655           24  2018-05-08T14:00:00.000Z   
35830           24  2018-05-08T14:00:00.000Z   
36003           24  2018-05-08T14:00:00.000Z   
36171           24  2018-05-08T14:00:00.000Z   
36340           24  2018-05-08T14:00:00.000Z   

                                                    tags    views   likes  \
32808  Netflix|"Trailer"|"Netflix Original Series"|"N...  2280153  106553   
32972  Netflix|"Trailer"|"Netflix Original Series"|"N...  4145678  166732   
33144  Netflix|"Trailer"|"Netflix Original Series"|"N...  5127308  193628   
33314  Netflix|"Trailer"|"Netflix Original Series"|"N...  5784740  209646   
33489  Netflix|"Trailer"|"Netflix Original Series"|"N...  6295801  221316   
33829  Netflix|"Trailer"|"Netflix Original Series"|"N...  6760593  22

In [17]:
#There are columns with inaccurate data types and we need to fix them
#change trending_date to datetime object
#change publish_time to datetime object

youtube['trending_date'] = pd.to_datetime(youtube['trending_date'], format="%y.%d.%m")
youtube['publish_time'] = pd.to_datetime(youtube['publish_time'])
youtube.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 38742 entries, 0 to 38915
Data columns (total 16 columns):
 #   Column                  Non-Null Count  Dtype              
---  ------                  --------------  -----              
 0   video_id                38742 non-null  object             
 1   trending_date           38742 non-null  datetime64[ns]     
 2   title                   38742 non-null  object             
 3   channel_title           38742 non-null  object             
 4   category_id             38742 non-null  int64              
 5   publish_time            38742 non-null  datetime64[ns, UTC]
 6   tags                    38742 non-null  object             
 7   views                   38742 non-null  int64              
 8   likes                   38742 non-null  int64              
 9   dislikes                38742 non-null  int64              
 10  comment_count           38742 non-null  int64              
 11  thumbnail_link          38742 non-null  o

In [18]:
#statistical summary

youtube.describe()

category_id         views         likes      dislikes  comment_count
count  38742.000000  3.874200e+04  3.874200e+04  3.874200e+04   3.874200e+04
mean      16.825667  5.911516e+06  1.344666e+05  7.606117e+03   1.308096e+04
std        7.752390  1.901316e+07  3.498336e+05  5.104749e+04   5.070138e+04
min        1.000000  8.510000e+02  0.000000e+00  0.000000e+00   0.000000e+00
25%       10.000000  2.509262e+05  5.881250e+03  2.000000e+02   6.770000e+02
50%       20.000000  9.799170e+05  2.516350e+04  8.200000e+02   2.475000e+03
75%       24.000000  3.675558e+06  1.140090e+05  3.353750e+03   9.222750e+03
max       43.000000  4.245389e+08  5.613827e+06  1.944971e+06   1.626501e+06

In [19]:
#save the dataset after processed
youtube.to_csv('youtube_edited.csv', index=False)

- The data cleaning has been completed.
- I have replace missing values, remove duplicates and make sure data types are correct.
- I have also save the dataset as csv file.
- Below, I did some data extraction

In [20]:
# maximum views, likes and dislikes for each category

youtube.groupby('category_id').agg({'views':'max','likes':'max','dislikes':'max'})

views    likes  dislikes
category_id                              
1             42560014   461135     37132
2              5362914   114841     12647
10           424538912  5613827    421473
15             6416920   187524      3461
17            22975455  1291139    193053
19             6362355    23791      2434
20            18158133  1031695     17093
22            62338362   625174    258504
23            43460605  1253154     38693
24           169884583  3312868   1944971
25            18994966   610526     37789
26             8721841   341145     13141
27            11240174   383049      8924
28            21530325   392249     19405
29            26703269  2061275    545595
43             1709880    44588       361

In [21]:
#which videos trending for more than a month

grouped = youtube.groupby('title')
grouped.filter(lambda x: len(x) > 30)

#another method
#youtube[youtube['title'].groupby(youtube['title']).transform('size')>10]


video_id trending_date  \
14196  pjTj-_55WZ8    2018-01-26   
14401  pjTj-_55WZ8    2018-01-27   
14603  pjTj-_55WZ8    2018-01-28   
14805  pjTj-_55WZ8    2018-01-29   
15015  pjTj-_55WZ8    2018-01-30   
...            ...           ...   
38909  Ibp6KqGRVbY    2018-06-14   
38911  l884wKofd54    2018-06-14   
38912  IP8k2xkhOdI    2018-06-14   
38913  Il-an3K9pjg    2018-06-14   
38914  -DRsfNObKIQ    2018-06-14   

                                                   title  \
14196  Rudimental - These Days feat. Jess Glynne, Mac...   
14401  Rudimental - These Days feat. Jess Glynne, Mac...   
14603  Rudimental - These Days feat. Jess Glynne, Mac...   
14805  Rudimental - These Days feat. Jess Glynne, Mac...   
15015  Rudimental - These Days feat. Jess Glynne, Mac...   
...                                                  ...   
38909  Ty Dolla $ign - Clout feat. 21 Savage [Lyric V...   
38911  Enrique Iglesias - MOVE TO MIAMI (Official Vid...   
38912  Jacob Sartorius - Up With It (Official Music V...   
38913                 Anne-Marie - 2002 [Official Video]   
38914  Eleni Foureira - Fuego - Cyprus - LIVE - First...   

                 channel_title  category_id              publish_time  \
14196               Rudimental           10 2018-01-25 12:00:16+00:00   
14401               Rudimental           10 2018-01-25 12:00:16+00:00   
14603               Rudimental           10 2018-01-25 12:00:16+00:00   
14805               Rudimental           10 2018-01-25 12:00:16+00:00   
15015               Rudimental           10 2018-01-25 12:00:16+00:00   
...                        ...          ...                       ...   
38909            Ty Dolla $ign           10 2018-05-09 16:27:39+00:00   
38911      EnriqueIglesiasVEVO           10 2018-05-09 07:00:01+00:00   
38912          Jacob Sartorius           10 2018-05-11 17:09:16+00:00   
38913               Anne-Marie           10 2018-05-08 11:05:08+00:00   
38914  Eurovision Song Contest           24 2018-05-08 20:32:32+00:00   

                                                    tags     views   likes  \
14196  rudimental|"rudimentaluk"|"official video"|"co...    228174   17376   
14401  rudimental|"rudimentaluk"|"official video"|"co...    826352   41379   
14603  rudimental|"rudimentaluk"|"official video"|"co...   1311780   53502   
14805  rudimental|"rudimentaluk"|"official video"|"co...   1728225   62178   
15015  rudimental|"rudimentaluk"|"official video"|"co...   2271446   73707   
...                                                  ...       ...     ...   
38909  Ty Dolla $ign|"Ty Dolla"|"Ty Dollar"|"Ty Dolla...   6078793   75335   
38911  Enrique Iglesias feat. Pitbull|"MOVE TO MIAMI"...  25066952  268088   
38912  jacob sartorius|"jacob"|"up with it"|"jacob sa...   1492219   61998   
38913  anne|"marie"|"anne-marie"|"2002"|"two thousand...  29641412  394830   
38914  Eurovision Song Contest|"2018"|"Lisbon"|"Cypru...  14317515  151870   

       dislikes  comment_count  \
14196       148            687   
14401       443           1047   
14603       618           1167   
14805       760           1306   
15015       923           1502   
...         ...            ...   
38909      2106           1269   
38911     12783           9933   
38912     13781          24330   
38913      8892          19988   
38914     45875          26766   

                                       thumbnail_link  comments_disabled  \
14196  https://i.ytimg.com/vi/pjTj-_55WZ8/default.jpg              False   
14401  https://i.ytimg.com/vi/pjTj-_55WZ8/default.jpg              False   
14603  https://i.ytimg.com/vi/pjTj-_55WZ8/default.jpg              False   
14805  https://i.ytimg.com/vi/pjTj-_55WZ8/default.jpg              False   
15015  https://i.ytimg.com/vi/pjTj-_55WZ8/default.jpg              False   
...                                               ...                ...   
38909  https://i.ytimg.com/vi/Ibp6KqGRVbY/default.jpg              False   
38911  https://i.ytimg.co

In [22]:
#which videos has views more than 1m but dislikes more than 500k

#events.loc[
    #(events["Sport"] == "Wrestling") & 
    #(events["Medal"] == "Gold"), "Nationality"]

youtube.loc[(youtube['views'] >= 1000000) & (youtube['dislikes'] >= 500000), ["title","views","dislikes"]]


title      views  dislikes
4794   YouTube Rewind: The Shape of 2017 | #YouTubeRe...   52611730    884948
4995   YouTube Rewind: The Shape of 2017 | #YouTubeRe...   75969469   1127802
5197   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  100911567   1353649
5399   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  113874632   1470384
5624   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  125432237   1545014
5838   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  137843120   1602383
6058   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  149376127   1643060
6280   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  156305907   1668460
6506   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  158002943   1683321
6713   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  159604206   1694945
6923   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  160544046   1704861
7132   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  161185315   1712284
7344   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  162929816   1722307
7563   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  163726971   1727826
7772   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  164743418   1732859
7978   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  165905535   1739579
8189   YouTube Rewind: The Shape of 2017 | #YouTubeRe...  169884583   1753274
9993                                           So Sorry.   13305605    629123
10193                                          So Sorry.   23389090   1065772
10397                                          So Sorry.   28407744   1278889
10605                                          So Sorry.   31675237   1415778
10811                                          So Sorry.   33998325   1517522
11015                                          So Sorry.   35983135   1611044
11229                                          So Sorry.   37539570   1674420
11492                                          So Sorry.   40610237   1802219
11719                                          So Sorry.   42028161   1843521
11928                                          So Sorry.   42829793   1868756
12134                                          So Sorry.   43596655   1890743
12343                                          So Sorry.   44266614   1911836
12559                                          So Sorry.   44815907   1928677
12784                                          So Sorry.   45349447   1944971
15663                         Suicide: Be Here Tomorrow.   24828198    509459
15880                         Suicide: Be Here Tomorrow.   25434829    519137
16093                         Suicide: Be Here Tomorrow.   26139495    535170
16310                         Suicide: Be Here Tomorrow.   26703269    545595